### PR TITLE
TTI: Check legalization cost of abs nodes

### PR DESCRIPTION
**Pull request by arsenm**

### CODE REVIEW

**Comment by arsenm** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/100523?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#100523** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/100523?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#100522** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/100522?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#100521** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/100521?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#100520** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/100520?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#100519** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/100519?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#100518** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/100518?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#100514** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/100514?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#100513** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/100513?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#97463** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/97463?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arsenm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: llvm/test/Analysis/CostModel/AMDGPU/abs.ll [hunk 1]
```diff
@@ -54,11 +54,11 @@ define i32 @abs_nonpoison(i32 %arg) {
...
-; FAST-NEXT:  Cost Model: Found an estimated cost of 70 for instruction: %V16I16 = call <16 x i16> @llvm.abs.v16i16(<16 x i16> undef, i1 false)
-; FAST-NEXT:  Cost Model: Found an estimated cost of 114 for instruction: %V17I16 = call <17 x i16> @llvm.abs.v17i16(<17 x i16> undef, i1 false)
-; FAST-NEXT:  Cost Model: Found an estimated cost of 174 for instruction: %V32I16 = call <32 x i16> @llvm.abs.v32i16(<32 x i16> undef, i1 false)
+; FAST-NEXT:  Cost Model: Found an estimated cost of 2 for instruction: %V4I16 = call <4 x i16> @llvm.abs.v4i16(<4 x i16> undef, i1 false)
+; FAST-NEXT:  Cost Model: Found an estimated cost of 2 for instruction: %V8I16 = call <8 x i16> @llvm.abs.v8i16(<8 x i16> undef, i1 false)
+; FAST-NEXT:  Cost Model: Found an estimated cost of 2 for instruction: %V16I16 = call <16 x i16> @llvm.abs.v16i16(<16 x i16> undef, i1 false)
+; FAST-NEXT:  Cost Model: Found an estimated cost of 2 for instruction: %V17I16 = call <17 x i16> @llvm.abs.v17i16(<17 x i16> undef, i1 false)
+; FAST-NEXT:  Cost Model: Found an estimated cost of 2 for instruction: %V32I16 = call <32 x i16> @llvm.abs.v32i16(<32 x i16> undef, i1 false)
```
Comment (jayfoad) [inline llvm/test/Analysis/CostModel/AMDGPU/abs.ll:61]: What is this demonstrating? 2 does not seem like the right cost for any VALU/SALU operation on v32i16.

Comment (RKSimon) [reply llvm/test/Analysis/CostModel/AMDGPU/abs.ll:61]: ISD::ABS v32i16 is set to CUSTOM lowering in SIISelLowering.cpp so the cost used is `LT.first * 2` - this assumption appears to be the cause of most regression in this patch series

**Review by RKSimon — APPROVED** [review]
LGTM

**Comment by arsenm** [comment]
### Merge activity

* **Aug 9, 4:27 AM EDT**: @arsenm started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/llvm/llvm-project/100523).

**Pull request merged**